src/guides/plugins/notebook.md

```diff
@@ -0,0 +1,464 @@
+# Notebook Plugin Development Guide
```

Comment:
I dont have much to add to this example, two comments:
- I would maybe add a screenshots where it makes sense to show what the code renders
- In general I would suggest that the description be less "textual" explanation of the code and more about the why/how the code does what it does.

Lets take the sentence: `- The `caido.storage.set()` method is called which updates the array with the new note.`
Storage set is so much more than that and IMO the rest of the other points are irrelevant to the reader since they can read the code. What they need to know is that before calling storage.set we are only operating in the frontend so if you close the app you lose the data, this calls the backend and **persists** the new note (you are actually missing an await there FYI). You also have to be aware that storage uses a `last write wins` strategy, meaning that if you add a note in another browser tab it will be overwritten unless you have a `storage.onChange` that syncs **your local storage**. 

A possible to explain that last concept is thinking in terms of "view of the world". In tab 1 "view of the world" I know about note A. In tab 2 "view of the world" I know about note A and B. The server "view of the world" also knows about note A and B. So if tab 1 "imposes" its "view of the world" to the server by adding note C then the server will take it and overwrite its "view of the world", discarding note B.

Reply:
Should I keep the current explanations for those who aren't as proficient at reading JS (such as me while making this plugin)? Haha.

Also, is that why I needed to make a copy of the array at the beginning? It would only persist through one application open and close before.

Reply:
Maybe inline comments in the code.

I didnt read the whole thing, we can take a time to look at it.

Reply:
With what you said:

"A possible to explain that last concept is thinking in terms of "view of the world". In tab 1 "view of the world" I know about note A. In tab 2 "view of the world" I know about note A and B. The server "view of the world" also knows about note A and B. So if tab 1 "imposes" its "view of the world" to the server by adding note C then the server will take it and overwrite its "view of the world", discarding note B."

By note C do you mean tab 1 writing its own note B?

Reply:
Yeah